tests/NLog.UnitTests/LogFactoryTests.cs

```diff
@@ -234,6 +234,31 @@ public void NewAttrOnNLogLevelShouldNotThrowError()
                 </rules>
             </nlog>");
         }
+
+        [Fact]
+        public void ValueWithVariableMustNotCauseInfiniteRecursion()
+        {
+            LogManager.Configuration = null;
+            
+            File.WriteAllText("NLog.config", @"
+            <nlog>
+                <variable name='dir' value='c:\mylogs' />
+                <targets>
+                    <target name='f' type='file' fileName='${var:dir}\test.log' />
+                </targets>
+                <rules>
+                    <logger name='*' writeTo='f' />
+                </rules>
+            </nlog>");
+            try
+            {
+                LogManager.Configuration.ToString();
```

Comment:
This is a good test! Can confirm it giving me a stackoverflow in the master! Great work!